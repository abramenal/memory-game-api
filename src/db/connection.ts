import knex from 'knex';
import camelcaseKeys from 'camelcase-keys';
import { snakeCase } from 'snake-case';

import { POSTGRES_HOST, POSTGRES_PORT, POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_DB, NODE_ENV } from '../bootstrap';

const isDevelopment = NODE_ENV === 'development';

export default knex({
  client: 'pg',
  debug: isDevelopment,
  connection: {
    host: POSTGRES_HOST,
    port: POSTGRES_PORT,
    database: POSTGRES_DB,
    user: POSTGRES_USER,
    password: POSTGRES_PASSWORD,
  },
  pool: {
    min: 2,
    max: 10,
  },
  migrations: {
    tableName: 'knex_migrations',
    directory: 'migrations',
  },
  postProcessResponse: result => {
    if (Array.isArray(result)) {
      return result.map(row => camelcaseKeys(row, { deep: true }));
    }

    return camelcaseKeys(result, { deep: true });
  },
  wrapIdentifier: (value, origImpl) => origImpl(snakeCase(value)),
});
