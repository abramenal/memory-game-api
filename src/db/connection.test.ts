import connection from './connection';

jest.mock('knex', () => (config: Record<string, unknown>) => config);
jest.mock('../bootstrap', () => ({
  POSTGRES_HOST: 'POSTGRES_HOST',
  POSTGRES_PORT: 'POSTGRES_PORT',
  POSTGRES_USER: 'POSTGRES_USER',
  POSTGRES_PASSWORD: 'POSTGRES_PASSWORD',
  POSTGRES_DB: 'POSTGRES_DB',
  NODE_ENV: 'NODE_ENV',
}));

describe('connection', () => {
  it('matches the snapshot', () => {
    expect(connection).toMatchSnapshot();
  });
});
