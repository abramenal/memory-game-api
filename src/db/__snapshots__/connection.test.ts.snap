// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connection matches the snapshot 1`] = `
Object {
  "client": "pg",
  "connection": Object {
    "database": "POSTGRES_DB",
    "host": "POSTGRES_HOST",
    "password": "POSTGRES_PASSWORD",
    "port": "POSTGRES_PORT",
    "user": "POSTGRES_USER",
  },
  "debug": false,
  "migrations": Object {
    "directory": "migrations",
    "tableName": "knex_migrations",
  },
  "pool": Object {
    "max": 10,
    "min": 2,
  },
  "postProcessResponse": [Function],
  "wrapIdentifier": [Function],
}
`;
